src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java

```diff
@@ -84,25 +83,51 @@ public BattleshipModel() {
 
         scanResult = false;
     }
-
+// Refactored getShip into a case statement instead of a mass of cluttered if statements. -GH
     public Ship getShip(String shipName) {
-        if (shipName.equalsIgnoreCase("aircraftCarrier")) {
-            return aircraftCarrier;
-        } if(shipName.equalsIgnoreCase("battleship")) {
-            return battleship;
-        } if(shipName.equalsIgnoreCase("submarine")) {
-            return submarine;
-        } if(shipName.equalsIgnoreCase("clipper")) {
-            return clipper;
-        }if(shipName.equalsIgnoreCase("dinghy")) {
-            return dinghy;
-        } else {
-            return null;
+        switch(shipName) {
```

Comment:
This switch statement looks a lot cleaner than the many if's you replaced. However, this approach does still use many return statements that would make debugging difficult if this wasn't such a simple method. Unfortunately adding an assignment and break statement to every case would make this look ugly compared to what you have. Nice work.